examples/flower_classifier/image_pyfunc.py

```diff
@@ -0,0 +1,165 @@
+        def decode_img(x):
+            return pd.Series(base64.decodebytes(bytearray(x[0], encoding="utf8")))
+
+        images = input.apply(axis=1, func=decode_img)
+        probs = self._predict_images(images)
+        m, n = probs.shape
+        label_idx = np.argmax(probs, axis=1)
+        labels = np.array([self._domain[i] for i in label_idx], dtype=np.str).reshape(m, 1)
+        output_data = np.concatenate((labels, label_idx.reshape(m, 1), probs), axis=1)
+        res = pd.DataFrame(columns=self._column_names, data=output_data)
+        res.index = input.index
+        return res
+
+    @staticmethod
+    def save_model(path, mlflow_model, keras_model, image_dims, domain):
+        """
+        Save a KerasImageClassifierPyfunc model to a path on the local file system.
+        :param path: Local path where the model is to be saved.
+        :param mlflow_model: :py:mod:`mlflow.models.Model` this flavor is being added to.
+        :param keras_model: Keras model to be saved.
+        :param image_dims: Image dimensions the Keras model expects.
+        :param domain: Labels for the classes this model can predict.
+        """
+        with TempDir() as tmp:
+            conf = {
+                "image_dims": "/".join(map(str, image_dims)),
+                "domain": "/".join(map(str, domain))
+            }
+            with open(tmp.path("conf.yaml"), "w") as f:
+                yaml.safe_dump(conf, stream=f)
+            keras_path = tmp.path("keras_model")
+            mlflow.keras.save_model(keras_model, path=keras_path)
+            conda_env = tmp.path("conda_env.yaml")
+            conda_env = _mlflow_conda_env(
+                path=conda_env,
+                additional_conda_deps=[
+                    "keras={}".format(keras.__version__),
+                    "{tf}=={version}".format(tf=tf.__name__, version=tf.__version__)
+                ],
+                additional_pip_deps=["pillow"],
+                additional_conda_channels=None,
+            )
+            mlflow.pyfunc.save_model(path,
+                                     __name__,
+                                     code_path=[__file__],
+                                     data_path=tmp.path("."),
+                                     model=mlflow_model,
+                                     conda_env=conda_env)
+
+    @classmethod
+    def log_model(cls, keras_model, artifact_path, image_dims, domain):
+        """
+        Log a KerasImageClassifierPyfunc model as an MLflow artifact for the current run.
+
+        :param keras_model: Keras model to be saved.
+        :param artifact_path: Run-relative artifact path this model is to be saved to.
+        :param image_dims: Image dimensions the Keras model expects.
+        :param domain: Labels for the classes this model can predict.
+        """
+        from mlflow.models import Model
+        Model.log(artifact_path=artifact_path,
+                  flavor=cls,
```

Comment:
Why did we create a new flavor? I thought we could just use the pyfunc flavor for this.